### ISSUE TITLE
path wrongly precedes globbing

### ISSUE THREAD

**Issue by danielshahaf** [zsh-users/zsh-syntax-highlighting#468]
```zsh
% touch \*
% echo *
```

The `*` is highlighted as `path`, it should be highlighted as `globbing`.

Bisected to dc1b2f6fa4bb477e7b9712ebdb4e82beb50258c6.

**Comment by phy1729** [comment]
In the case of path overlapping with other styles, which comes first? (We should probably document this decision somewhere). For example
```
touch foo\*
echo foo'*'
```
should that be `(6 11 path)` or `(6 8 path 9 11 single-quoted-argument)` or something else?

**Comment by danielshahaf** [comment]
I'm leaning towards the former.

**Comment by danielshahaf** [comment]
There's a third option: `("6 8 underline" "9 11 fg=yellow,underline")` (assuming the default styles).

**Comment by phy1729** [comment]
I lean towards the second option. The first seems less useful in the case of `*-quoted-argument-unclosed` and the third would require a driver API change (afact without writing code). Do you have an idea on how to implement the third option or were you just listing it for completeness sake?

**Comment by danielshahaf** [comment]
My reasoning for preferring the first to the second is that users are accustomed to an underlined thing (not at the end of LBUFFER) being a complete filename; inserting a yellow '' partway breaks that, and I don't think the cases that benefit from it are too common.

This is really another way of saying that the first option was the status quo up to last month, and the change away from it wasn't intentional.

**Comment by phy1729** [comment]
The status quo was `'f'oo` would be highlighted as `single-quoted-argument`. Currently (bdbe214) `'f'oo` is highlighted as `path`. Arguably the second would be closer to what the status quo was.

I think the status quo with respect to globbing isn't ideal. Personally I'd prefer that the globbing style apply only to the characters which are special in a globbing sense not the whole word.

**Comment by phy1729** [comment]
Although `fo'o'` was `path`, so who knows which side is closer to status quo (or if it's even desirable to maintain).

**Comment by danielshahaf** [comment]
How about if both `fo'o'` and `'f'oo` are highlighted as path?

**Comment by danielshahaf** [comment]
There are three questions here:

1. The original post
2. `'f'oo` as path or string
3. `foo*bar` what extent should globbing apply to

Could we please break the third (and possibly the second) to separate issues?  This thread is becoming hard to follow.

**Comment by phy1729** [comment]
Third should certainly be split (will do in a bit). I think the second is necessary to know how to fix the ordering.

**Comment by danielshahaf** [comment]
The current incarnation of #473, 75c1ca06dff372e849f1539ea6a936f8c201d5c9, highlights `'f'oo` as `'f'` yellow+underline and `oo` in default color + underline.
